tsdb/ooo_head.go

```diff
@@ -113,13 +120,15 @@ type OOORangeHead struct {
 	// the timerange of the query and having preexisting pointers to the first
 	// and last timestamp help with that.
 	mint, maxt int64
+	tombstones tombstones.Reader
 }
 
 func NewOOORangeHead(head *Head, mint, maxt int64) *OOORangeHead {
 	return &OOORangeHead{
-		head: head,
-		mint: mint,
-		maxt: maxt,
+		head:       head,
+		mint:       mint,
+		maxt:       maxt,
+		tombstones: head.tombstones,
```

Comment:
`head.tombstones` will only span the range of the in-order samples, as when doing a delete request, the time range is clamped to the min/max times for in-order samples (`head.MinTime()`/`head.MaxTime()` only keeps track of in-order samples):

https://github.com/prometheus/prometheus/blob/9825a80498de18183cdded8074c2a7c908182428/tsdb/head.go#L1465-L1466

We'd need to update the logic to take into account of the OOO time range too.